netket/vqs/mc_expect_grad.py

```diff
@@ -373,3 +393,58 @@ def gradfun(der_loc_vals, der_logs_ave):
         LdagL_grad,
         model_state,
     )
+
+
+@partial(jax.jit, static_argnums=(0, 1))
+def _grad_expect_continuous(
+    model_apply_fun: Callable,
+    kernel,
+    parameters: PyTree,
+    model_state: PyTree,
+    x: jnp.ndarray,
+) -> Tuple[PyTree, PyTree]:
+
+    x_shape = x.shape
+    if jnp.ndim(x) != 2:
+        x = x.reshape((-1, x_shape[-1]))
+
+    n_samples = x.shape[0] * mpi.n_nodes
+
+    def logpsi(w, σ):
+        return model_apply_fun({"params": w, **model_state}, σ)
+
+    local_value_vmap = jax.vmap(
+        partial(kernel, logpsi),
+        in_axes=(None, 0),
+        out_axes=0,
+    )
+    x = x.reshape((-1, 10, x_shape[-1]))
+
+    def compute_kernel(i, x):
+        Oloc = local_value_vmap(parameters, x)
+
+        return i, Oloc
+
+    _, O_loc = jax.lax.scan(compute_kernel, 0, x)
```

Comment:
why are you using a scan here?

Reply:
computing the kinetic energy needs  a lot of memory because of the second order derivative. So this was a try to reduce memory consumption. Instead of feeding all samples at once I feed each sample individually (or in this case 10 samples a time to be precise)